### ISSUE TITLE
Create and review an Experiment document workflow diagram

### ISSUE THREAD

**Issue by eguil** [ES-DOC/esdoc-docs#136]
Create workflow diagram for Experiments (including Experimental requirements):
- Add UID and versioning in workflows
- Fit in CIM documents
- Review (ME)

**Comment by charliepascoe** [comment]
[Cmip6_experiments_workflow_2.pdf](https://github.com/ES-DOC/esdoc-docs/files/212796/Cmip6_experiments_workflow_2.pdf)

**Comment by charliepascoe** [comment]
Updated workflow to give a broad view of who is responsible for what ES-Doc vs MIP Teams.
![cmip6_experiments_workflow_01 283 29](https://cloud.githubusercontent.com/assets/3264143/14425960/30be59ec-ffe3-11e5-8c27-fe43cbbe5ae0.jpg)

**Comment by charliepascoe** [comment]
Also as pdf
[Cmip6_experiments_workflow_6.pdf](https://github.com/ES-DOC/esdoc-docs/files/212854/Cmip6_experiments_workflow_6.pdf)

**Comment by murphysj** [comment]
On the 4/11/16 esdoc call we agreed that the workflow diagrams exist initially to help us understand all the steps involved in preparing for cmip6.

**Comment by murphysj** [comment]
WRT External QC, I think we need a review by Karl et al., in addition to a QC by each group just looking at their MIP. Karl or someone at that level can review the entire suite and scrub for conflicting names etc.

**Comment by murphysj** [comment]
Do we need to plan for modifications AFTER CMIP6 has started and data is arriving?

**Comment by murphysj** [comment]
How will the Experiment documents in the archive interact with the data archive?

**Comment by charliepascoe** [comment]
![slide1](https://cloud.githubusercontent.com/assets/3264143/14826241/d9b40630-0bd4-11e6-8445-872ad5746aae.jpg)

**Comment by charliepascoe** [comment]
Guide to the colours I've used for the boxes on the CMIP6 experiment workflow diagram
Red: Tasks 
Turquoise: External input
Grey: Decisions
Green: Identifiers
Blue: Publication

**Comment by charliepascoe** [comment]
Workflow Text

Various documents have been created to collate information about the CMIP6 experiments, most notably the Overview of CMIP6 endorsed MIPs which contains a suite of proposals from each of the MIPs that will be contributing to CMIP6. http://wcrp-climate.org/images/modelling/WGCM/CMIP/CMIP6-EndorsedMIPs_Summary_150819_Sent.pdf

The information about the experiments described in the Overview of CMIP6 endorsed MIPs is cross referenced with other WCRP documents and then codified into CIM structures using a spreadsheet tool.  The MIP PIs may also be contacted at this stage to help with any unresolved queries.  The CMIP6 experiments spreadsheets are kept in the ES-Doc github repository. https://github.com/ES-DOC/esdoc-docs/tree/master/cmip6/experiments

The CMIP6 experiments spreadsheets are rendered as CIM documents and published as “draft” status on the es-doc.org website.  The draft content on es-doc.org is reviewed by the ES-Doc team and recommendations are made as to improving the spreadsheet content and/or the rendering of the information in the spreadsheets.

The MIP teams and members of the WCRP are invited to review the “CMIP6-DRAFT” metadata on http://search.es-doc.org once the internal review process is satisfied that the information contained in the spreadsheets are sufficiently rendered and the content is OK.   Any changes recommended by the external review process are made via 
adjustments to the CMIP6 experiments spreadsheets.

Once the external reviewers are satisfied the CIM documents for the CMIP6 experiments are published as “public” status on the es-doc.org website and issued with a version ID. If errors are found or updates are necessary after public publication the ES-Doc team amend the CMIP6 experiments spreadsheet as appropriate.

At the time of writing it is undecided whether permanent document IDs should be added to the CMIP6 experiments spreadsheet directly or generated by the spreadsheet rendering code.

**Comment by charliepascoe** [comment]
Updated Workflow Text

Various documents have been created to collate information about the CMIP6 experiments, most notably the Overview of CMIP6 endorsed MIPs which contains a suite of proposals from each of the MIPs that will be contributing to CMIP6. http://wcrp-climate.org/images/modelling/WGCM/CMIP/CMIP6-EndorsedMIPs_Summary_150819_Sent.pdf

The information about the experiments described in the Overview of CMIP6 endorsed MIPs is cross referenced with other WCRP documents and then codified into CIM structures using a spreadsheet tool. The MIP PIs may also be contacted at this stage to help with any unresolved queries. The CMIP6 experiments spreadsheets are kept in the ES-Doc github repository. https://github.com/ES-DOC/esdoc-docs/tree/master/cmip6/experiments

The CMIP6 experiments spreadsheets are rendered as CIM documents and published with “draft” status on the es-doc.org website.  The minor version number of the CIM documents are either issued or updated at this stage. 

The draft content on es-doc.org is reviewed by the ES-Doc team and recommendations are made as to improving the spreadsheet content and/or the rendering of the information in the spreadsheets.

The MIP teams and members of the WCRP are invited to review the “CMIP6-DRAFT” metadata on http://search.es-doc.org once the internal review process is satisfied that the information contained in the spreadsheets are sufficiently rendered and the content is OK. Any changes recommended by the external review process are made via 
adjustments to the CMIP6 experiments spreadsheets.

Once the external reviewers are satisfied, the CIM documents for the CMIP6 experiments are published with "public” status on the es-doc.org website. The major version number of the CIM documents are either issued or updated at this stage. 

If errors are found or updates are necessary after public publication the ES-Doc team amend the CMIP6 experiments spreadsheet as appropriate.

At the time of writing it is undecided whether permanent document IDs should be added to the CMIP6 experiments spreadsheet directly or generated by the spreadsheet rendering code.

**Comment by eguil** [comment]
Comments by Mark E.:

Hi Charlotte/Eric

I took a look at the workflow diagram on github and the process looks fine.  A few related comments to the work flow and the wider ES-DOC system:
 
-          Rough dates would be useful at least for the main transitions for the MIP team review and the final publication
-          This may have already been decided – but given there are three sources for experiment names – what is the agreement for which of these will be the master.  I assume that Karl’s list was only a temporary list while he was discussing issues with the MIPs so this will not be maintained.  What plans are there for keeping dreq and esdoc in step – I hope that the big changes are all done now, but there will be a number of smaller changes in coming months.
-          In doing your analysis of experiment requirements you indentified several areas where there was a possibility of rationalisation (e.g. simulation periods).  This may not be very important for MIP teams but is absolutely CRITICAL to the modelling groups.  I can already see that the change the CMIP6 autonomous “MIP” approach is introducing complexities for the modelling teams that are hard to deal with and the plethora of slightly different experiment constraints is only going to make things worse.  If this rationalisation is not done at a central level it will be done by each modelling group (because they need to standardise their model configurations) – with the result that the data supplied from each modelling group will be unnecessarily different.  The workflow should deal explicitly with this rationalisation – I’m not saying that it has to be an ES-DOC responsibility.
-          It is still not clear to me who is responsible for the fixed enumerations (e.g. like MIP name, model name, model type code, insititute code, forcing codes, etc.).  If this is ES-DOC – how does it fit into the workflow.  If it is not ES-DOC who is responsible?.  In either case how will modelling groups get access to the latest versions of these enumerations.  I hope  this is covered in Eric’s ES-DOC document.
-          We are working with Mark Greenslade on the export of experiment / requirements for modelling groups that are going to use pyesdoc for delivering metadata.  I’m still a little unsure how to map your requirement/constraint structure into our metadata schema, but once we get some example outputs from Mark I’ll be able to try and get my head round it with real examples.
-          The mechanism of receiving the model properties and creating our model documents using pyesdoc seems reasonably clear.  The remaining issues – which I hope will be resolved in Eric’s document - are:
§  At the last ES-DOC meeting there was some discussion about having to deliver references and contacts as individual documents (one per reference/contact) before we use them in the model descriptions.  For pyesdoc users this is not going to be easy to manage.  Some thought needs to be given to the pyesdoc workflow for managing this.
§  Our goal is to try and deliver the model documents as early as we can – hopefully before we deliver any data. In CMIP5 we had late updates to the model properties in the questionnaire requiring us to add information and republish our documents on two occasions.  Some thought needs to be given to the pyesdoc users workflow.  We can get updated model property lists – but what will be provided to help us decide which properties are new, which properties have been removed and which properties have different configuration (e.g. permitted values)
-          Apart from the issues in (6) I think I have a reasonable understanding of how we are going to use exports of model property CVs and pyesdoc to deliver our model metadata, but I am still very unclear on how we will deliver CIM documents on experiment conformance, initialisation and forcing.  Again I assume this will all be explained in Eric’s document.

My current (very high level) understanding is this:
§  We deliver data sets to an ESGF node and an automated process within the ingest/publish activity will generate a CIM simulation document for that dataset.  [Not clear to me whether that gets updated if we update that dataset at a later date (adding more years, and/or variables) – or for some reason the atomic data sets (variable file sets) within a single simulation are ingested at different times by the ESGF node.]

§  We use pyesdoc to create and deliver one or more CIM documents that provide ensemble level information that reference the automatically generated simulation document(s)

Some questions I have are:

§  What are the CIM documents that we need to generate to cover the ensemble – where will this be defined.

§  How do we get the reference to the automatically generated simulation document – will this be a pyesdoc function (i.e we pass something like MIP, experiment, model, realisation and it returns the document reference if it has it)

§  What happens if after delivering the initial set of simulations for an ensemble, we decide to extend the delivery with one or more elements.  Can we just resend another complete.

I’ll happily volunteer to review Eric’s document and I promise to be a little more responsive than I was with the workflow ;-0

Regards

mark
